safe/impact_function_v4/impact_function.py

```diff
@@ -702,28 +743,84 @@ def debug_layer(self, layer, check_fields=True):
         return name
 
     def run(self):
-        """Run the whole impact function."""
-
-        self.validate()
-
-        self.reset_state()
-        clear_prof_data()
-        self._run()
-
-        # Get the profiling log
-        self._performance_log = profiling_log()
-        self.callback(8, 8, analysis_steps['profiling'])
-
-        self._profiling_table = create_profile_layer(
-            self.performance_log_message())
-        _, name = self.debug_layer(self._profiling_table)
-        self._profiling_table = self.datastore.layer(name)
-        check_inasafe_fields(self._profiling_table)
+        """Run the whole impact function.
+
+        :return: A tuple with the status of the IF and an error message if
+            needed.
+            The status is 0 if everything was fine.
```

Comment:
Perhaps we can use a variable for the status code. For example:
SUCCESS = 1, INPUT_PROBLEM = 2, INASAFE_PROBLEM = 3 etc and put them like in definitions module.